internal/driver/config_test.go

```diff
@@ -51,3 +54,56 @@ func TestUpdateFromRaw(t *testing.T) {
 		})
 	}
 }
+
+// TestGetCameraXAddr verifies the parsing of the camera XAddr
+func TestGetCameraXAddr(t *testing.T) {
+
+	tests := []struct {
+		input         map[string]models.ProtocolProperties
+		expected      string
+		errorExpected bool
+	}{
+		{
+			input: map[string]models.ProtocolProperties{
+				OnvifProtocol: {
+					Address: "http://localhost",
+					Port:    "80",
+				},
+			},
+			expected: "http://localhost:80",
+		},
+		{
+			input: map[string]models.ProtocolProperties{
+				OnvifProtocol: {
+					CustomMetadata: "custommetadata",
+				},
+			},
+			errorExpected: true,
+		},
+		{
+			input: map[string]models.ProtocolProperties{
+				OnvifProtocol: {
+					Address: "",
+					Port:    "",
+				},
+			},
+			errorExpected: true,
+		},
+		{
+			errorExpected: true,
+		},
+	}
```

Comment:
Also you can add a test case where the port is only empty and no error is expected.